src/app/components/List/EditionsList.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router';
+import EBookList from './EBookList';
+
+const EditionsList = ({ work, eReaderUrl, max }) => {
+  const list = work.instances;
+  if (!list || list.length === 0) {
+    return null;
+  }
+
+  const getPublisher = (instance) => {
+    let publisher = instance && instance.agents && instance.agents.find(agent => agent.roles.indexOf('publisher') > -1);
+    publisher = publisher ? publisher.name : null;
+    return publisher;
+  };
+  const getIsValid = instance => (instance.items && instance.items.length > 0)
+    || (instance.pub_date && instance.pub_date_display)
+    || instance.pub_place
+    || getPublisher(instance);
+
+  const filterValid = instances => instances.filter(l => getIsValid(l));
```

Comment:
Why not use `instance` instead of `l` to be more explicit?

Reply:
Sure, fixed.